ftdc/custom_format.go

```diff
@@ -131,13 +135,20 @@ func isNumeric(kind reflect.Kind) bool {
 
 func flattenStruct(item reflect.Value) ([]float32, error) {
 	flattenPtr := func(inp reflect.Value) reflect.Value {
-		for inp.Kind() == reflect.Pointer {
+		for inp.Kind() == reflect.Pointer || inp.Kind() == reflect.Interface {
+			if inp.IsNil() {
+				return inp
+			}
+
 			inp = inp.Elem()
 		}
 		return inp
 	}
 
 	rVal := flattenPtr(item)
+	if (rVal.Kind() == reflect.Pointer || rVal.Kind() == reflect.Interface) && rVal.IsNil() {
```

Comment:
What is this predicate saying? If a flattened pointer is still a pointer or interface and that pointer is nil then return `[]float32{}`? Makes sense to me, but double-checking my understanding.